demo/tsconfig.json

```diff
@@ -23,6 +23,7 @@
         "baseUrl": ".",
         "paths": {
             "*": [
+                "./node_modules/tns-core-modules/*",
```

Comment:
The AoT compilation in Groceries (as well as the other NS apps, maintained by the team) is used only with webpack. When bundling with webpack, we use the [tsconfig.aot.json](https://github.com/NativeScript/sample-Groceries/blob/master/tsconfig.aot.json) configuration for TypeScript. The paths, specified in the tsconfig.aot.json, are different from the ones in `tsconfig.json`. I'll call them `full paths` from now on. Using the full paths helps prevent the memory problem when bundling. I wrote more about that in the already mentioned issue - https://github.com/angular/angular-cli/issues/5618#issuecomment-306479219.
You may ask why don't we use the same approach for plugins. I think that adding the full tsconfig paths in all plugins is not a sustainable solution. After all, we may add new folders to `tns-core-modules`, change the names of the existing ones or even drop some of them. If that happens, the plugin authors will have to go over all their plugins' tsconfigs and update them one by one or they won't be able to compile them with tsc anymore.
But what about the apps that have the full paths? Well, as I mentioned, they are placed in the `tsconfig.aot.json` file - and this one is added by the nativescript-dev-webpack plugin. Then, everyone that has that file, uses the plugin. If a change is required in the full paths, we can use the ns-webpack to modify the `tsconfig.aot.json` file (on postinstall, for example) and fix the user's project.
After all, we hope that the memory problem when using `@ngtools/webpack` gets resolved soon. But before that happens, I think it's best to recommend for plugin authors to use full imports - `tns-core-modules/...`.